### PR TITLE
Allow opt out of container styling, renderBackdrop

### PR DESCRIPTION
**Pull request by jquense**

### CODE REVIEW

File: src/Modal.js [hunk 1]
```diff
@@ -212,7 +229,7 @@ const Modal = React.createClass({
       onExiting,
       onEnter,
       onEntering,
-      onEntered,
```
Comment (taion) [inline src/Modal.js]: o7 dangling comma

File: src/Modal.js [hunk 2]
```diff
@@ -340,7 +365,7 @@ const Modal = React.createClass({
     let doc = ownerDocument(this);
     let container = getContainer(this.props.container, doc.body);
 
-    modalManager.add(this, container, this.props.containerClassName);
```
Comment (taion) [inline src/Modal.js]: what did this use to do?

File: src/ModalManager.js [hunk 3]
```diff
@@ -20,21 +20,48 @@ function findContainer(data, modal) {
...
+    style.paddingRight =
+      parseInt(css(container, 'paddingRight') || 0, 10) + getScrollbarSize() + 'px';
+  }
+
+  css(container, style);
+}
+
+function removeContainerStyle({ style }, container) {
+
+  Object.keys(style).forEach(
```
Comment (taion) [inline src/ModalManager.js]: `Object.assign(container.style, style)`?

might need to pull in that Babel transform though

File: src/Modal.js (hunk 2 above)
Comment (jquense) [inline src/Modal.js]: same thing, I just moved the default instance to a defaultProp so you can configure them via the component

**Comment by jquense** [comment]
there is a bunch of clean up work we need to do here at somepoint
not sure waht the deal is with the Affix tests btw

**Pull request merged**